core/blocksigner/blocksigner.go

```diff
@@ -53,6 +53,7 @@ func New(pub ed25519.PublicKey, hsm Signer, db pg.DB, c *protocol.Chain) *BlockS
 func (s *BlockSigner) SignBlock(ctx context.Context, b *bc.Block) ([]byte, error) {
 	sig, err := s.hsm.Sign(ctx, s.Pub, &b.BlockHeader)
 	if err != nil {
+		//TODO(ameets): rethink error pkg to preserve full stack trace
```

Comment:
I think this TODO should be removed. It's not about this PR or this line of code.

(Also note that judicious use of `errors.Wrap`/`Wrapf` is even better than full stack traces, since they let us preserve exactly the important information and leave out the chaff.)